module-gui/gui/widgets/text/core/cursors/TextBlockCursor.cpp

```diff
@@ -367,7 +367,7 @@ namespace gui
 
     auto BlockCursor::begin() -> std::list<TextBlock>::iterator
     {
-        return document == nullptr ? document->blocks.end() : document->blocks.begin();
+        return document->blocks.begin();
```

Comment:
consider either assert for non existing document or rising an exception

Reply:
Added asserts